selectors/drafts.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {createSelector} from 'reselect';
+
+import {getMyChannels} from 'mattermost-redux/selectors/entities/channels';
+import {GlobalState} from 'types/store';
+import {PostDraft} from 'types/store/rhs';
+
+import {StoragePrefixes} from 'utils/constants';
+
+type Info = {
+    id: string;
+    type: 'channel' | 'thread';
+}
+
+export type Draft = Info & {
+    key: keyof GlobalState['storage']['storage'];
+    value: PostDraft;
+    timestamp: Date;
+};
+
+export type DraftSelector = (state: GlobalState) => Draft[];
+export type DraftCountSelector = (state: GlobalState) => number;
+
+function getInfoFromKey(key: string, prefix: string): Info {
+    const keyArr = key.split('_');
+    if (prefix === StoragePrefixes.DRAFT) {
+        return {
+            id: keyArr[1],
+            type: 'channel',
+        };
+    }
+
+    return {
+        id: keyArr[2],
+        type: 'thread',
+    };
+}
+
+export function makeGetDraftsByPrefix(prefix: string): DraftSelector {
+    return createSelector(
+        'makeGetDraftsByPrefix',
+        (state: GlobalState) => state.storage?.storage,
+        (storage) => {
+            if (!storage) {
+                return [];
+            }
+
+            return Object.keys(storage).flatMap((key) => {
+                const item = storage[key];
+                if (
+                    key.startsWith(prefix) &&
+                    item !== null &&
+                    item.value != null &&
+                    (item.value.message || item.value.fileInfos.length > 0)
```

Comment:
(0/5, may be more readable being explicit) We can remove the `> 0`, since length will be always >= 0 and 0 will be considered as false.

That being said, how much can we trust that fileInfos will always be an array (will never be undefined) and we will always have length? I guess we can trust, but wanted to make the question either way.